tidy3d/plugins/mode/mode_solver.py

```diff
@@ -71,6 +82,30 @@ def freqs_not_empty(cls, val):
             raise ValidationError("ModeSolver 'freqs' must be a non-empty tuple.")
         return val
 
+    @pydantic.validator("compute_group_index")
+    def assign_default_on_true(cls, val):
+        """Assing the default fractional frequency step value if not provided."""
+        if val is True:
+            return GROUP_INDEX_STEP
+        return val
+
+    def _post_init_validators(self):
+        if self.compute_group_index >= 1:
+            raise ValidationError("Argument 'compute_group_index' must be less than 1.")
+
+        if self.compute_group_index > 0:
+            if self.mode_spec.track_freq is None:
+                log.warning(
```

Comment:
I think it would improve usability if we could tell the user specifically what field to set in these warnings. For example, by adding the sentence `Consider setting the 'ModeSpec.track_freq' field to __.` Same for the precision. `Consider setting the 'ModeSpec.precision' field to 'double'`.